xdsl/transforms/riscv_scf_loop_range_folding.py

```diff
@@ -7,68 +7,73 @@
     RewritePattern,
     op_type_rewrite_pattern,
 )
+from xdsl.transforms.canonicalization_patterns.riscv import get_constant_value
 
 
 class HoistIndexTimesConstant(RewritePattern):
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: riscv_scf.ForOp, rewriter: PatternRewriter) -> None:
         index = op.body.block.args[0]
 
-        uses: list[riscv.MulOp] = []
```

Comment:
the main thing that's changed is that now we don't handle multiple constants. This is because we now have riscv-cse, which will eliminate the multiplications by the same constant anyway